servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -448,13 +449,22 @@ public DefaultSingleAddressHttpClientBuilder<U, R> appendConnectionFilter(
         requireNonNull(factory);
         connectionFilterFactory = appendConnectionFilter(connectionFilterFactory, factory);
         strategyComputation.add(factory);
+        ifHostHeaderHttpRequesterFilter(factory);
         return this;
     }
 
     @Override
     public DefaultSingleAddressHttpClientBuilder<U, R> appendConnectionFilter(
             final Predicate<StreamingHttpRequest> predicate, final StreamingHttpConnectionFilterFactory factory) {
-        return appendConnectionFilter(toConditionalConnectionFilterFactory(predicate, factory));
+        appendConnectionFilter(toConditionalConnectionFilterFactory(predicate, factory));
+        ifHostHeaderHttpRequesterFilter(factory);
+        return this;
+    }
+
+    private void ifHostHeaderHttpRequesterFilter(final Object filter) {
```

Comment:
do you mean `isHostHeaderHttpRequesterFilter`? I would normally write this as returning a `boolean` and do the assignment to a final field in constructor.

Reply:
It's used in 4 different places, the goal was to write a single method that will take care of `addHostHeaderFallbackFilter` variable. Open for other name suggestions.

Reply:
I'm always a sucker for pure functions rather than side-effect functions and if a field can be made final, I'll go to extra trouble. Maybe not terribly important here.